### PR TITLE
GUI: Allow opening BIF files directly

### PR DESCRIPTION
**Pull request by michaelpm54**
Let me know if you want any changes of course or if it should be done a different way.

### CODE REVIEW

File: src/aurora/keyfile.cpp [hunk 1]
```diff
@@ -202,4 +202,18 @@ Common::SeekableReadStream *KEYFile::getResource(uint32 index, bool UNUSED(tryNo
 	return iRes.dataFile->getResource(iRes.resIndex);
 }
 
+std::vector<Archive::Resource *> KEYFile::getResourceListForDataFile(Common::UString dataFile) {
```
Comment (DrMcCoy) [inline src/aurora/keyfile.cpp]: That should take a `const Common::UString &`.

And the return could be a `std::vector<const Archive::Resource *>`

The method itself is const, too.

File: src/gui/resourcetree.cpp [hunk 2]
```diff
@@ -205,15 +225,23 @@ bool ResourceTree::hasChildren(const QModelIndex &index) const {
 	return itemFromIndex(index)->hasChildren();
 }
 
-void ResourceTree::insertItemsFromArchive(Archive &archive, const QString &path,
+void ResourceTree::insertItemsFromArchive(Archive &archive, ResourceTreeItem *item,
```
Comment (DrMcCoy) [inline src/gui/resourcetree.cpp]: That can be a `const ResourceTreeItem &`, no?

**Comment by michaelpm54** [comment]
Updated.
- Added const where requested and replaced pointer with ref.
- Added an early return [here](https://github.com/xoreos/phaethon/pull/30/commits/1b891616752928febdd8c91686fe14b1babc128b#diff-d586d5c0a96afc76f19d4078c059e6e8R197) which should prevent adding items twice.

**Comment by DrMcCoy** [comment]
You forgot making `KEYFile::getResourceListForDataFile()` itself const.

**Comment by michaelpm54** [comment]
Good eye, fixed.

**Comment by DrMcCoy** [comment]
Merged, thanks! :)

**Pull request merged**